### ISSUE TITLE
Many to Many Issue

### ISSUE THREAD

**Issue by ashkan18** [hyperoslo/Sync#84]
I had issues with Many to Many relationships. I found the problem happens at this line:
```
childPredicate = [NSPredicate predicateWithFormat:@"ANY %K.%K = %@", relationshipName, destinationLocalKey, [children valueForKey:destinationRemoteKey]];
```
I noticed that we generate a predicate that points to the child but since it has relationshipName in it, it ends up being invalid predicate. 

Here are the models diagram:
![models](https://cloud.githubusercontent.com/assets/1230819/7274747/79c8eee6-e8cd-11e4-9526-de105941d8bf.png)

The predicate that that line generates is 
```
ANY fulfillers.xid = (<some xids>)
```
But it does it on fulfillers Model. 

Here is a JSON sample:
```json
{
  "xid":"mstaff_F58dVBTsXznvMpCPmpQgyV",
  "user":{
    "full_name":"Ashkan Nasseri",
    "email":"a@b.com",
    "phone":"###-###-####",
    "xid":"usr_8YQhMfiLvXEg6TaD7YKyjL",
    "photo_inventory":true
  },
  "image":"a.jpg",
  "image_xs":"s.jpg",
  "fulfillers":[
    {
      "xid":"ffr_AkAHQegYkrobp5xc2ySc5D",
      "name":"New York"
    },
    {
      "xid":"ffr_n5eGjfHQRqKr4tAfL7RNi9",
      "name":"Chicago"
    }
  ],
  "current_vehicle":{
    "xid":"vhc_test",
    "name":"Test"
  }
}
```
 
I fixed the problem by removing %k. and relationshipName from the predicate generation, everything works fine for me now but it breaks testCustomPrimaryKey.

**Comment by 3lvis** [comment]
@ashkan18 Hi!

If you are posting about a crash in your application or a feature request, an example of your JSON and your Core Data model or a stacktrace would be really helpful for us to be able to reproduce your issue or understand your request.

**Comment by ashkan18** [comment]
Sorry I was still writing the issue that I accidentally submitted :)

**Comment by 3lvis** [comment]
Thanks for explaining the issue is such detail, I'll look into it.

**Comment by 3lvis** [comment]
Hi @ashkan18, everything seems to be working fine for me, I tried your JSON and your Model and the new tests are passing.

https://github.com/hyperoslo/Sync/pull/90

I wonder if it's something that's been fixed in Sync `1.0.3`.

Would you mind verifying? Thanks!

**Comment by ashkan18** [comment]
@3lvis Works like a charm! Thanks man and great work!

**Comment by 3lvis** [comment]
Thanks! 🍻
![giphy](https://cloud.githubusercontent.com/assets/1088217/7319728/e2f9e9d0-ea97-11e4-8dd5-3486ccbd8627.gif)

**Comment by ashkan18** [comment]
hey @3lvis so our tests with latest code are passing but when I tried it in the actual app I ended up seeing the same issue again, I tracked down the issue and it seems the predicate i was talking about crashes on DATAStackSQLiteStoreType and it works fine when we use DATAStackInMemoryStoreType.

That kind of exactly matches the giphy you posted :)

**Comment by ashkan18** [comment]
in your test if you change line 19 in your Test.m to switch to SQLiteStore you'll see the test fails:
```Objective-C
DATAStack *dataStack = [[DATAStack alloc] initWithModelName:modelName
                                                         bundle:[NSBundle bundleForClass:[self class]]
                                                      storeType:DATAStackSQLiteStoreType];
```

**Comment by 3lvis** [comment]
@ashkan18 Woah, that's weird. I was able to reproduce it. Will have a look :)

**Comment by 3lvis** [comment]
Hi again @ashkan18! I took some time to work in this issues, I found a few issues with using SQLiteStore that should be fixed in `Sync 1.0.4`, let me know how it goes.

https://github.com/hyperoslo/Sync/releases/tag/1.0.4

**Comment by ashkan18** [comment]
Thanks @3lvis ! works great!
